File: inference-api/inference_api_typescript_client/src/sample.ts
// sample.ts
import 'source-map-support/register';
import { InferenceAPIClient } from './client';

async function main() {
  const username = 'datashare-admin';
  const privateKeyFilename = '/Users/gabealvarez/bunkerhill/bunkerhill/private_key.pem';
  const baseUrl = 'https://api.coppshillhealth.com/';

  const client = new InferenceAPIClient(username, privateKeyFilename, undefined, baseUrl);

  const modelId = 'e7ad8122-14cf-4bb2-b57f-075a07b51e2b';
  const patientMrn = '1';
  
  try {
    const inferences = await client.getInferencesAsync(
      modelId,
      patientMrn,
    );
    
    console.log(JSON.stringify(inferences, null, 2));
  } catch (error) {
    console.error(`Error while getting inferences: ${error}`);
  }
}

main();

File: client_sdk/python/sample.py
import asyncio

from bunkerhill_inference_api import InferenceAPIClient


async def main():
    async with InferenceAPIClient(
        username="datashare-admin",
        private_key_filename="private_key.pem",
        base_url="https://api.coppshillhealth.com/",
    ) as client:
        inference_list = await client.get_inferences(
            model_id="e7ad8122-14cf-4bb2-b57f-075a07b51e2b",
            patient_mrn="1",
        )
        print(inference_list)


asyncio.run(main())
